src/test/java/com/programmers/springbootbasic/domain/userVoucherWallet/infrastructure/JdbcUserVoucherWalletRepositoryTest.java

```diff
@@ -0,0 +1,389 @@
+package com.programmers.springbootbasic.domain.userVoucherWallet.infrastructure;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.assertj.core.api.Assertions.tuple;
+import static org.junit.jupiter.api.Assertions.assertNotEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+
+import com.programmers.springbootbasic.common.TimeGenerator;
+import com.programmers.springbootbasic.domain.user.domain.UserRepository;
+import com.programmers.springbootbasic.domain.user.domain.entity.User;
+import com.programmers.springbootbasic.domain.TestTimeGenerator;
+import com.programmers.springbootbasic.domain.userVoucherWallet.domain.UserVoucherWalletRepository;
+import com.programmers.springbootbasic.domain.userVoucherWallet.domain.entity.UserVoucherWallet;
+import com.programmers.springbootbasic.domain.voucher.domain.ProdVoucherIdGenerator;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherIdGenerator;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherRepository;
+import com.programmers.springbootbasic.domain.voucher.domain.VoucherType.FixedAmountVoucher;
+import com.programmers.springbootbasic.domain.voucher.domain.entity.Voucher;
+import java.util.List;
+import java.util.Random;
+import java.util.UUID;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.dao.DataAccessException;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.transaction.annotation.Transactional;
+
+@SpringBootTest
+@Transactional
+@ActiveProfiles("test")
```

Comment:
```suggestion
@JdbcTest
@ActiveProfiles("test")
@ContextConfiguration(classes = {
        JdbcUserVoucherWalletRepository.class,
        JdbcUserRepository.class,
        JdbcVoucherRepository.class
})
@AutoConfigureTestDatabase(replace = AutoConfigureTestDatabase.Replace.NONE)
```

테스트 코드가 갈수록 늘어나게 되는건 불가피합니다.
따라서 테스트 코드의 실행 시간에 해해서 고민을 하게 되는데요.
가장 큰 원인이 모든 테스트에서 @SpringBootTest를 사용하는 것입니다.
전체 context가 필요한 바깥쪽 (Controller) 테스트 영역이 있고, 안으로 들어올 수록 필요한 context가 줄어들게 됩니다.
현재 repository 레이어에서는 @SpringBootTest가 꼭 필요 없어보입니다.